### PR TITLE
Operator Enhancement

### PR DESCRIPTION
**Pull request by MUzairS15**
**Description**

This PR fixes #
1. Initialise logging for meshery-operator.
2. Ensures the operator redeploys the controllers, if they are deleted or their status isn't Running (k8s world status)
3. Adds health check for MeshSync

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by MUzairS15** [comment]
TODO: Replace logging with meshkit logger, redefine errors to be meshkit-errors.

**Comment by MUzairS15** [comment]
https://github.com/user-attachments/assets/575ebdd0-8593-4ea1-9165-4588d3bc51e7

File: controllers/broker_controller.go [hunk 1]
```diff
@@ -67,7 +68,9 @@ func (r *BrokerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	// Check if Broker controller deployed
 	result, err := r.reconcileBroker(ctx, true, baseResource, req)
 	if err != nil {
-		return ctrl.Result{}, ErrReconcileBroker(err)
+		err = ErrReconcileBroker(err)
+		r.Log.Error(err, "broker reconcilation failed")
```
Comment (leecalcote) [inline controllers/broker_controller.go:72]: Reconciliation failures means what?

File: controllers/broker_controller.go [hunk 2]
```diff
@@ -79,18 +82,24 @@ func (r *BrokerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	// Get broker endpoint
 	err = brokerpackage.GetEndpoint(ctx, baseResource, r.Clientset, r.KubeConfig.Host)
 	if err != nil {
-		return ctrl.Result{}, ErrGetEndpoint(err)
+		err = ErrGetEndpoint(err)
+		r.Log.Error(err, "unable to get the broker endpoint")
```
Comment (leecalcote) [inline controllers/broker_controller.go:86]: meaning that the endpoint is undefined or that a network-based attempt failed?

File: controllers/broker_controller.go [hunk 3]
```diff
@@ -79,18 +82,24 @@ func (r *BrokerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
...
+		return ctrl.Result{}, err
 	}
 
 	// Patch the broker resource
 	patch, err := utils.Marshal(baseResource)
 	if err != nil {
-		return ctrl.Result{}, ErrUpdateResource(err)
+		err = ErrUpdateResource(err)
+		r.Log.Error(err, "unable to update broker resource")
```
Comment (leecalcote) [inline controllers/broker_controller.go:94]: Meaning that the resource couldn't be found or that it is present but the operator doesn't have permission or something else?

File: controllers/broker_controller.go [hunk 4]
```diff
@@ -79,18 +82,24 @@ func (r *BrokerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
...
+		return ctrl.Result{}, err
 	}
 
 	err = r.Status().Patch(ctx, baseResource, client.RawPatch(types.MergePatchType, []byte(patch)))
 	if err != nil {
-		return ctrl.Result{}, ErrUpdateResource(err)
+		err = ErrUpdateResource(err)
+		r.Log.Error(err, "unable to update broker resource")
```
Comment (leecalcote) [inline controllers/broker_controller.go:101]: Meaning that the configuration was rejected from Kube API or that the resource isn't present or that the requested configuration was missing or invalid or what?

File: controllers/error.go [hunk 5]
```diff
@@ -39,53 +39,53 @@ const (
 
 // Error definitions
 func ErrGetMeshsync(err error) error {
-	return errors.New(ErrGetMeshsyncCode + ":" + "Unable to get meshsync resource")
+	return errors.New(ErrGetMeshsyncCode + ":" + "Unable to get meshsync resource" + err.Error())
```
Comment (leecalcote) [inline controllers/error.go:42]: Meaning that the custom resource was never created?

**Review by leecalcote — CHANGES_REQUESTED** [review]
As a user troubleshooting Meshery deployment, I'm not confident that I would know what to do when seeing these errors.

**Comment by leecalcote** [comment]
@innocentrda, your feedback here is helpful.

**Comment by leecalcote** [comment]
Is there an adjoining addition to the https://docs.meshery.io/guides/troubleshooting/meshery-operator-meshsync guide?

**Comment by innocentrda** [comment]
> As a user troubleshooting Meshery deployment, I'm not confident that I would know what to do when seeing these errors.

That's true. It's a good improvement to see those errors but being a bit detailed on why those errors and what to do can be helpful.

**Pull request merged**